core/helpers/getAndCallAsyncDefault.ts

```diff
@@ -0,0 +1,10 @@
+interface DefaultExport {
+  default: Function
+}
+type LoaderFunc = () => Promise<DefaultExport>
+type WrappedFunc = (...args) => Promise<any>
+
+export default (loader: LoaderFunc): WrappedFunc => async function (...args) {
+  const service = await loader();
+  return service.default(...args);
+}
```

Comment:
can you add generic types?
```
type WrappedFunc<T, R> = (...args: T) => Promise<R>

export default (loader: LoaderFunc)<T, R>: WrappedFunc<T, R> => async function (...args) {
  const service = await loader();
  return service.default(...args);
}
```
I'm not sure if this is working example, but you know what i mean :P

I know that we don't have types for helpers right now, but maybe someone else would like to use it with types :shrug: 
`export const setGroupedProduct = getAndCallAsyncDefault<product, product>` etc